apps/src/storage/firebaseStorage.js

```diff
@@ -502,6 +504,32 @@ FirebaseStorage.copyStaticTable = function(tableName, onSuccess, onError) {
     .then(onSuccess, onError);
 };
 
+function enforceUniqueTableNames(tableName) {
+  let projectRef = getProjectDatabase();
+  return Promise.all([
+    projectRef
+      .child(`counters/tables/${tableName}`)
+      .once('value')
+      .then(snapshot => {
+        if (snapshot.val()) {
+          return Promise.reject(
+            'There is already a table with name ' + tableName
+          );
+        }
+      }),
+    projectRef
```

Comment:
Looks like these two sections (510-519 and 520-529) are only different by the child value. ('counters/table' and 'current_table'). Mind moving them to a function?

Reply:
Done